fw/http_frame.c

```diff
@@ -1967,23 +1973,30 @@ tfw_h2_insert_frame_header(struct sock *sk, TfwH2Ctx *ctx, TfwStream *stream,
 	length = tfw_h2_calc_frame_length(ctx, stream, type, len,
 					  max_len - FRAME_HEADER_SIZE);
 	if (type == HTTP2_DATA) {
-		ctx->rem_wnd -= length;
-		stream->rem_wnd -= length;
+		if (!stream->xmit.t_len) {
```

Comment:
This is mistake, we should adjust window in case when response contains trailers.
I implement new test for this case.
https://github.com/tempesta-tech/tempesta-test/pull/651
It passes on master but fails on your branch

Reply:
I thought we could adjust the wnd stuff until trailers handling, but yes, you're right, maybe just revert them.